### PR TITLE
Payments/Carriers preferences applied also on non PS carriers

### PR DESCRIPTION
**Pull request by vincentbz**
| Questions | Answers |
| --- | --- |
| Branch? | deveop |
| Description? | In Payments preferences, Carrier restrictions can be applied only on PS carriers. It adds the other ones |
| Type? | bug fix |
| Category? | BO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-1323 |
| How to test? | Add a carrier module and see if you can set that restriction on it |

### CODE REVIEW

File: controllers/admin/AdminPaymentPreferencesController.php [hunk 1]
```diff
@@ -258,7 +258,7 @@ public function renderView()
                           'identifier' => 'id_country',
                           'icon' => 'icon-globe',
                     ),
-                    array('items' => Carrier::getCarriers($this->context->language->id),
+                    array('items' => Carrier::getCarriers($this->context->language->id, false, false, false, null, (defined('ALL_CARRIERS') ? ALL_CARRIERS : null)),
```
Comment (kpodemski) [inline controllers/admin/AdminPaymentPreferencesController.php]: we shouldn't have here ALL_CARRIERS but Carrier::ALL_CARRIERS, and we don't need to check if that's defined

Comment (vincentbz) [inline controllers/admin/AdminPaymentPreferencesController.php]: Done, thanks :)

**Comment by vincentbz** [comment]
Damn it, I put a wrong commit in that PR.

**Comment by aleeks** [comment]
@vincentbz we will see it ahah !

**Comment by vincentbz** [comment]
Better here : https://github.com/PrestaShop/PrestaShop/pull/6538

**Pull request closed without merging**